internal/fs/fs.go

```diff
@@ -157,7 +157,11 @@ func NewFileSystem(
 	// enabled only if cache-dir is not empty and file-cache:max-size-mb is non 0.
 	var fileCacheHandler *file.CacheHandler
 	if config.IsFileCacheEnabled(cfg.MountConfig) {
-		fileCacheHandler = createFileCacheHandler(cfg)
+		var err error
+		fileCacheHandler, err = createFileCacheHandler(cfg)
+		if err != nil {
+			return nil, fmt.Errorf("while creating NewFileSystem(): %v", err)
```

Comment:
I think it's fine to simply return the error instead of reformatting it because createFileCacheHandler is not used more than once.